cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -217,13 +217,33 @@ func getAPIServerCommand(cfg *kubeadmapi.MasterConfiguration) []string {
 			defaultArguments["audit-log-maxage"] = fmt.Sprintf("%d", *cfg.AuditPolicyConfiguration.LogMaxAge)
 		}
 	}
-
+	if cfg.APIServerExtraArgs == nil {
+		cfg.APIServerExtraArgs = map[string]string{}
+	}
+	cfg.APIServerExtraArgs["authorization-mode"] = getAuthzModes(cfg)
 	command = append(command, kubeadmutil.BuildArgumentListFromMap(defaultArguments, cfg.APIServerExtraArgs)...)
-	command = append(command, getAuthzParameters(cfg.AuthorizationModes)...)
 
 	return command
 }
 
+// getAuthzModes gets the authorization-related parameters to the api server
+// At this point, we can assume the list of authorization modes is valid (due to that it has been validated in the API
+// machinery code already) If the list is empty; it's defaulted (mostly for unit testing)
+func getAuthzModes(cfg *kubeadmapi.MasterConfiguration) string {
```

Comment:
As you don't need all of `cfg` here I'd specify exactly what you need, i.e. `extraAuthzModes map[string]string`

Reply:
I deleted the code which was taking into account cfg.AuthorizationMode, but you mentioned its out of scope right now. So I kept the interface.